### PR TITLE
Remove upper-bound limit on ActiveRecord

### PR DESCRIPTION
**Pull request by Edouard-chin**
cc/ @wvanbergen @rafaelfranca

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,18 +8,24 @@ rvm:
...
 
 gemfile:
-- Gemfile.activerecord40
-- Gemfile.activerecord41
-- Gemfile.activerecord42
-- Gemfile.activerecord50
-- Gemfile.activerecord51
+  - gemfiles/Gemfile.activerecord40
+  - gemfiles/Gemfile.activerecord41
+  - gemfiles/Gemfile.activerecord42
```
Comment (wvanbergen) [inline .travis.yml]: I think we can safely stop testing against these old versions ^^^

File: .travis.yml [hunk 2]
```diff
@@ -3,23 +3,19 @@ language: ruby
 cache: bundler
 
 rvm:
-- 2.1.8
-- 2.2.4
 - 2.3.0
+- 2.4.1
```
Comment (rafaelfranca) [inline .travis.yml]: We need ruby 2.5 too.

File: activerecord-databasevalidations.gemspec [hunk 3]
```diff
@@ -18,7 +18,7 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ["lib"]
 
-  spec.add_runtime_dependency "activerecord", ">= 4.0", "< 5.2"
+  spec.add_runtime_dependency "activerecord", ">= 4.0"
```
Comment (rafaelfranca) [inline activerecord-databasevalidations.gemspec]: If we don't test, we also do support anymore, so we need to bump this up to 5.0

File: activerecord-databasevalidations.gemspec [hunk 4]
```diff
@@ -18,7 +18,7 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ["lib"]
```
Comment (rafaelfranca) [inline activerecord-databasevalidations.gemspec]: Also need to set the minimum ruby version here

Comment (Edouard-chin) [reply activerecord-databasevalidations.gemspec]: Thanks!

File: activerecord-databasevalidations.gemspec [hunk 5]
```diff
@@ -18,7 +18,9 @@ Gem::Specification.new do |spec|
...
 
-  spec.add_runtime_dependency "activerecord", ">= 4.0", "< 5.2"
+  spec.required_ruby_version = '>= 2.3.0'
```
Comment (rafaelfranca) [inline activerecord-databasevalidations.gemspec]: Any reason why 2.2 is not supported anymore?

Comment (Edouard-chin) [reply activerecord-databasevalidations.gemspec]: for some reason I was sure rails 5.0 required at least 2.3. Looks like I was wrong

**Pull request merged**